phonetest.py

```diff
@@ -200,18 +201,93 @@ def lookup(cls, phoneid, config_file, chunk):
     @classmethod
     def match(cls, tests=None, test_name=None, phoneid=None,
               config_file=None, job_guid=None,
-              build_url=None):
+              build_url=None, branch=None, revision=None):
```

Comment:
As mentioned below, I'd like to do the lookup outside of match and just pass a set of changed files that can be used to see if there is an intersection with the set of changed paths the test may require.